File: app/src/main/java/com/example/casino/fragment/mainFragment.java
package com.example.casino.fragment;

import android.os.Bundle;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.NavController;
import androidx.navigation.NavOptions;
import androidx.navigation.fragment.NavHostFragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.Toast;

import com.example.casino.MainActivity;
import com.example.casino.R;
import com.example.casino.viewmodel.CasinoViewModel;

public class mainFragment extends Fragment {

    private String mUsername;
    private String mPassword;

    private CasinoViewModel userViewModel;

    public static mainFragment newInstance() {
        mainFragment fragment = new mainFragment();
        Bundle args = new Bundle();
        fragment.setArguments(args);
        return fragment;
    }

    public mainFragment() {
        // Required empty public constructor
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_main,
                container, false);

        MainActivity mainActivity = (MainActivity)getActivity();
        userViewModel = mainActivity.getUserViewModel();
        Process_View_Login_Register_LiveData();
        Process_View_Button(view);


        return view;
    }

    private void Process_View_Login_Register_LiveData(){
        userViewModel.getMessageLoginRegisterLiveData().observe(getViewLifecycleOwner(), new Observer<String>() {
            @Override
            public void onChanged(String str) {
                if (str == "LOGIN SUCCESS") {
                    navigateToClientMenuFragment();
                } else if (str == "LOGIN FALSE") {
                    Toast toast = Toast.makeText(requireActivity(), "Login Fail", Toast.LENGTH_SHORT);
                    toast.show();
                } else if (str == "REGISTER SUCCESS") {
                    navigateToClientMenuFragment();
                } else if (str == "REGISTER FALSE") {
                    Toast toast = Toast.makeText(requireActivity(), "Register Fail", Toast.LENGTH_SHORT);
                    toast.show();
                } else {

                }
            }
        });
    }

    private void Process_View_Button(View view) {
        // Button login
        Button btnLogin = (Button) view.findViewById(R.id.button_login);
        btnLogin.setOnClickListener(new View.OnClickListener()
        {
            @Override
            public void onClick(View v)
            {
                userViewModel.Process_EI_Button_Login(mUsername, mPassword);
            }
        });

        // Button register
        Button btnRegister = (Button) view.findViewById(R.id.button_register);
        btnRegister.setOnClickListener(new View.OnClickListener()
        {
            @Override
            public void onClick(View v)
            {
                userViewModel.Process_EI_Button_Register(mUsername, mPassword);
            }
        });
    }

    private void navigateToClientMenuFragment(){
        NavController navController = NavHostFragment.findNavController(this);
        navController.navigate(
                R.id.action_mainFragment_to_clientMenuFragment,
                null,
                new NavOptions.Builder()
                        .setEnterAnim(android.R.animator.fade_in)
                        .setExitAnim(android.R.animator.fade_out)
                        .build()
        );
    }
}

File: app/src/main/java/com/example/casino/screens/mainFragment.kt
package com.example.casino.screens

import android.os.Bundle
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Button
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.livedata.observeAsState
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.ComposeView
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.unit.dp
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import androidx.navigation.NavController
import com.example.casino.viewmodel.CasinoViewModel

class MainFragment : Fragment() {

    private lateinit var userViewModel: CasinoViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        userViewModel = ViewModelProvider(requireActivity()).get(CasinoViewModel::class.java)
    }

    override fun onCreateView(
        inflater: android.view.LayoutInflater,
        container: android.view.ViewGroup?,
        savedInstanceState: Bundle?
    ): android.view.View? {
        return ComposeView(requireContext()).apply {
            setContent {
                MainFragmentContent(userViewModel)
            }
        }
    }
}

@Composable
fun MainFragmentContent(userViewModel: CasinoViewModel) {
    val message = userViewModel.messageLoginRegisterLiveData.observeAsState(initial = "")

    LaunchedEffect(message.value) {
        when (message.value) {
            "LOGIN SUCCESS", "REGISTER SUCCESS" -> {
                //navigateToClientMenuFragment()
            }
            "LOGIN FALSE", "REGISTER FALSE" -> {
                //Toast.makeText(LocalContext.current, "Login/Register Failed", Toast.LENGTH_SHORT).show()
            }
        }
    }

    Column(modifier = Modifier.padding(16.dp)) {
        Text(text = "Welcome to Casino", modifier = Modifier.padding(bottom = 16.dp))

        Button(onClick = {
            userViewModel.Process_EI_Button_Login("username", "password")
        }) {
            Text("Login")
        }

        Button(onClick = {
            userViewModel.Process_EI_Button_Register("username", "password")
        }) {
            Text("Register")
        }
    }
}

@Composable
fun navigateToClientMenuFragment() {
    val navController = LocalContext.current as? NavController
    navController?.navigate("client_menu_screen") {

    }
}
